NullGuard.Fody/ImplicitModeAnalyzer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Mono.Cecil;
+
+public class ImplicitModeAnalyzer : INullabilityAnalyzer
+{
+    public bool AllowsNull(PropertyDefinition property)
+    {
+        return property.ImplicitAllowsNull();
+    }
+
+    public bool AllowsNullInput(ParameterDefinition parameter, MethodDefinition method)
+    {
+        return parameter.ImplicitAllowsNull();
+    }
+
+    public bool AllowsNullOutput(ParameterDefinition parameter, MethodDefinition method)
+    {
+        // Maintain legacy behavior in non-NRT modes which does not check ref output values
+        if (!parameter.IsOut)
+            return true;
```

Comment:
@tom-englert If you guys decide you want to change the "legacy" behavior of NullGuard attributes applied to `ref` parameters so that it also checks the outgoing value then all you have to do is remove these lines here and in the explicit mode analyzer.